chia/consensus/blockchain.py

```diff
@@ -859,8 +865,21 @@ def seen_compact_proofs(self, vdf_info: VDFInfo, height: uint32) -> bool:
         self._seen_compact_proofs.add(pot_tuple)
         return False
 
+    class BlockInfo(Protocol):
+        @property
+        def prev_header_hash(self) -> bytes32:
+            pass
+
+        @property
+        def transactions_generator(self) -> Optional[SerializedProgram]:
+            pass
+
+        @property
+        def transactions_generator_ref_list(self) -> List[uint32]:
+            pass
```

Comment:
perhaps this is controversial. We really don't need a full block here, just these 3 features of a block. This makes it a lot simpler to test and benchmark this function

Reply:
I need to review this in detail but I rather not add stuff to this file

Reply:
I moved this protocol out into its own file, `chia/types/block_protocol.py`